code-collab-app/src/editorpage.js

```diff
@@ -6,39 +6,22 @@ import logo from './codeCollabLogo.png';
 import './App.css';
 import IconButton from '@material-ui/core/IconButton';
 import AddIcon from '@material-ui/icons/Add';
+import { inject, observer } from 'mobx-react';
 
 class Editorpage extends React.Component {
   constructor(props){
     super(props);
 
     this.state = {
-      cellCount: 0,
-      cellText: (this.props.location.data) ? this.props.location.data : [],
-      cellLocked: [],
-      cellCurrentlyEditing: null,
+      cellCount: this.props.peer_data.cell_count,
+      cellText: this.props.peer_data.doc_data,
+      cellLocked: this.props.peer_data.cell_locked,
+      cellCurrentlyEditing: this.props.peer_data.current_cell,
     }
   }
 
   componentDidMount() {
-    if(this.state.cellText.length === 0){
-      this.addEditorCell('print("Hello World!")');
-    }
-    this.props.location.peer.listen_for_req();
-  }
-
-  setCellText = (keyvalue, cellContents) => {
```

Comment:
Replaced this with a method on the peer_data side, and now all the changes are made on the peer_data side and those changes trickle back to editor page's state by props.